S7.Net/PLC.cs

```diff
@@ -229,28 +366,33 @@ public ErrorCode Open()
                         return ErrorCode.WrongCPU_Type;
                 }
 
+                //COTP Setup
                 _mSocket.Send(bSend1, 22, SocketFlags.None);
-                if (_mSocket.Receive(bReceive, 22, SocketFlags.None) != 22)
+                var response = ReadCOTP_TPDU();
+                if (response.PDUType != 0xd0) //Connect Confirm
                 {
                     throw new Exception(ErrorCode.WrongNumberReceivedBytes.ToString());
-                } 
+                }
 
+                //S7ComSetup
                 byte[] bsend2 = { 3, 0, 0, 25, 2, 240, 128, 50, 1, 0, 0, 255, 255, 0, 8, 0, 0, 240, 0, 0, 3, 0, 3, 1, 0 };
                 _mSocket.Send(bsend2, 25, SocketFlags.None);
 
-                if (_mSocket.Receive(bReceive, 27, SocketFlags.None) != 27)
+                var responseLength = ReciveCOTP_TSDU(bReceive);
```

Comment:
Same here check for 27 before? or is it fine to go into the ReciveCOTP_TSDU

Reply:
Yup 😣